client/src/Components/RepoBrowser/__tests__/RepoBrowser.test.tsx

```diff
@@ -0,0 +1,103 @@
+import { describe, expect, test } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import RepoBrowser from '@/Components/RepoBrowser/RepoBrowser';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import React from 'react';
+import nock from 'nock';
+import '@testing-library/jest-dom';
+
+import { configureStore } from '@reduxjs/toolkit';
+import { BrowserRouter as Router } from 'react-router-dom';
+import connectionReducer from '@/Store/Reducers/ConnectionReducer';
+import { Provider } from 'react-redux';
+import { FeatureFlagsProvider } from '@/Components/FeatureFlagsContext/FeatureFlagsContext';
+import { DataSource } from '@/api/Models';
+
+describe('Test RepoBrowser', () => {
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        staleTime: Infinity,
+      },
+    },
+  });
+
+  const store = configureStore({
+    reducer: {},
+  });
+
+  const AllProviders = ({ children }) => (
```

Comment:
Might be worth putting this in a shared setup file so that any components that use query client can utilise it